lib/doorkeeper/oauth/refresh_token_request.rb

```diff
@@ -51,12 +53,16 @@ def create_access_token
           client
         )
 
-        @access_token = AccessToken.create!(
+        create_params = {
           application_id: refresh_token.application_id,
+          expires_in: expires_in,
+          previous_refresh_token: refresh_token.refresh_token,
```

Comment:
First token starts with `refresh_token: nil`.
I think thats on purpose, no need to prevent it.